include/video/gazebocamerainput.h

```diff
@@ -60,13 +61,14 @@ class GazeboCameraInput : public Input
         // Subscribe to the topic, and register a callback
         m_ImageSub = m_ImageNode->Subscribe(topic, &GazeboCameraInput::OnImageMsg, this);
         LOG_INFO << "Subsribed to "<< topic <<"\n";
+        m_HaveReceivedFrameSemaphore.wait();
```

Comment:
This is giving me a compile error:
```
In file included from /home/shinjan/Workspaces/BrainsOnBoard/bob_robotics/examples/gazebo_tank/gazebo_tank.cc:8:0:
/home/shinjan/Workspaces/BrainsOnBoard/bob_robotics/cmake/../include/video/gazebocamerainput.h: In member function ‘virtual cv::Size BoBRobotics::Video::GazeboCameraInput::getOutputSize() const’:
/home/shinjan/Workspaces/BrainsOnBoard/bob_robotics/cmake/../include/video/gazebocamerainput.h:75:47: error: passing ‘const BoBRobotics::Semaphore’ as ‘this’ argument discards qualifiers [-fpermissive]
         m_HaveReceivedFrameSemaphore.waitOnce();
                                               ^
```
I think it's because `getOutputSize()` is marked as const.

Reply:
The solution to this is to mark the semaphore as ``mutable``, meaning that the semaphore's non-const methods can be called from const methods in ``GazeboCameraInput``

Reply:
Cool. I've incorporated this into the latest commit.